src/storage/compression/compression.cpp

```diff
@@ -73,13 +74,97 @@ uint32_t getDataTypeSizeInChunk(const common::PhysicalTypeID& dataType) {
     }
 }
 
+ALPMetadata::ALPMetadata(const alp::state& alpState, common::PhysicalTypeID physicalType)
+    : exp(alpState.exp), fac(alpState.fac), exceptionCount(alpState.exceptions_count) {
+    const size_t physicalTypeSize =
+        (physicalType == common::PhysicalTypeID::DOUBLE ? EncodeException<double>::sizeBytes() :
+                                                          EncodeException<float>::sizeBytes());
+
+    // to get the exception capacity we find the number of bytes needed to store the current
+    // exception count, take the smallest power of 2 greater than or equal to that value
+    // or the size of one page (whichever is larger)
+    // then find how many exceptions fit in that size
+    exceptionCapacity =
+        static_cast<uint64_t>(std::bit_ceil(alpState.exceptions_count * physicalTypeSize)) /
+        physicalTypeSize;
+}
+
+const CompressionMetadata& CompressionMetadata::getChild(offset_t idx) const {
+    return children[idx];
+}
+
+size_t CompressionMetadata::serializedSizeBytes(common::PhysicalTypeID internalDataType) {
+    size_t sizeToWrite = sizeof(min) + sizeof(max) + sizeof(compression);
+    if (internalDataType == common::PhysicalTypeID::FLOAT ||
+        internalDataType == common::PhysicalTypeID::DOUBLE) {
+        sizeToWrite += sizeof(ALPMetadata);
+    }
+    for (const auto& childType : getChildTypes(internalDataType)) {
+        sizeToWrite += serializedSizeBytes(childType);
+    }
+    return sizeToWrite + (8 - sizeToWrite % 8) % 8;
```

Comment:
I could remove this, I just included it since the `CompressionMetadata` struct was previously padded. I think I had previously removed it then added it back in while debugging.